tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContent.scala

```diff
@@ -23,16 +24,22 @@ class EncryptedEmailContentFactory(encrypter: Encrypter) {
     encrypt(preview.getValue)
 
   def encryptAttachmentMetadata(parsedAttachments: List[ParsedAttachment], messageId: MessageId): Option[String] = {
-    if (parsedAttachments.isEmpty) return None
-    Some(encrypt(AttachmentMetaDataSerializer
-      .serializeList(AttachmentMetadata.fromJava(parsedAttachments, messageId))
-      .toString()))
+    parsedAttachments match {
+      case Nil | List() => None
```

Comment:
Nil is not enough?

Reply:
Right, Not enough, If `parsedAttachments` is List(), We can still json deserialize that, so `encryptAttachmentMetadata` will has value